agent/lib/src/external_api/model.rs

```diff
@@ -0,0 +1,139 @@
+use serde::{Deserialize, Serialize};
+
+#[derive(Deserialize, Serialize, Debug, PartialEq, Eq)]
+pub struct FileModel {
+    pub filename: String,
+    pub content: String,
+}
+
+#[derive(Deserialize, Serialize, Debug)]
+pub struct CodeEntry {
+    pub files: Vec<FileModel>,
+    pub script: Vec<String>, // All commands to execute at startup
+}
+
+#[derive(Deserialize, Serialize, Debug, PartialEq, Eq)]
+pub enum Type {
+    Status,
+    Request,
+    Response,
+}
+
+#[derive(Deserialize, Serialize, Debug, PartialEq, Eq)]
+pub enum Code {
+    Run,
+    Ok,
+}
+
+#[derive(Deserialize, Serialize, Debug)]
+pub struct StatusMessage {
+    pub r#type: Type,
+    pub code: Code,
+}
+
+impl StatusMessage {
+    pub fn new() -> StatusMessage {
+        StatusMessage {
+            r#type: Type::Status,
+            code: Code::Ok,
+        }
+    }
+}
+
+impl Default for StatusMessage {
+    fn default() -> Self {
+        Self::new()
+    }
+}
```

Comment:
You don't need to implement the `new()` method then. Just use `Default` on this message.